src/Adapters/Files/GridFSBucketAdapter.js

```diff
@@ -68,29 +69,81 @@ export class GridFSBucketAdapter extends FilesAdapter {
     const stream = await bucket.openUploadStream(filename, {
       metadata: options.metadata,
     });
-    if (this._encryptionKey !== null) {
+
+    return new Promise((resolve, reject) => {
       try {
-        const iv = crypto.randomBytes(16);
-        const cipher = crypto.createCipheriv(this._algorithm, this._encryptionKey, iv);
-        const encryptedResult = Buffer.concat([
-          cipher.update(data),
-          cipher.final(),
-          iv,
-          cipher.getAuthTag(),
-        ]);
-        await stream.write(encryptedResult);
-      } catch (err) {
-        return new Promise((resolve, reject) => {
-          return reject(err);
-        });
+        const iv = this._encryptionKey !== null
+          ? crypto.randomBytes(16)
+          : null;
+
+        const cipher = this._encryptionKey !== null && iv
+          ? crypto.createCipheriv(this._algorithm, this._encryptionKey, iv)
+          : null;
+
+        // when working with a Blob, it could be over the max size of a buffer, so we need to stream it
+        if (typeof Blob !== 'undefined' && data instanceof Blob) {
```

Comment:
```suggestion
        if (data instanceof Blob) {
```
See https://github.com/parse-community/parse-server/pull/9286#discussion_r1791863516